pandas/_libs/src/ujson/python/date_conversions.c

```diff
@@ -90,7 +90,32 @@ char *PyDateTimeToIso(PyObject *obj, NPY_DATETIMEUNIT base,
 
     *len = (size_t)get_datetime_iso_8601_strlen(0, base);
     char *result = PyObject_Malloc(*len);
-    ret = make_iso_8601_datetime(&dts, result, *len, base);
+    // Check to see if PyDateTime has a timezone.
+    // Don't convert to UTC if it doesn't.
+    int is_tz_aware = 0;
+    PyObject *tmp;
+    PyObject *offset;
+    if (PyObject_HasAttrString((PyObject*)obj, "tzinfo")) {
```

Comment:
Can you remove the cast?

Reply:
could cast to PyDateTime or whatever up-front and use something like _PyDateTime_HAS_TZINFO instead of hasttr/getattr?

Reply:
Will try to use datetime C API in followup. Unfortunately, this PR is getting a little large already.